### PR TITLE
Update the Login Functionality

### PR DESCRIPTION
**Pull request by anmoldtu**

### CODE REVIEW

File: backend/auth/auth.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	"github.com/golang-jwt/jwt/v5"
+)
+
+func IsUCSDEmail(email string) bool {
+	// Check if the email has the "ucsd.edu" domain
+	return strings.HasSuffix(email, "ucsd.edu")
+}
+
+func VerifyJWTToken(c *gin.Context, token string) {
+	jwksURL := "https://www.googleapis.com/oauth2/v3/certs"
```
Comment (gurnoor6) [inline backend/auth/auth.go:18]: Let's add this in constants

**Review by Yamini-L — APPROVED** [review]
Looks great!

File: backend/login/login.go [hunk 2]
```diff
@@ -1,74 +1,96 @@
...
+	Picture string `json:"picture"`
 }
 
-const oauthStateString = "random"
+func getConfig() Config {
+	return Config{
+		ClientID:        os.Getenv("GOOGLE_CLIENT_ID"),
+		ClientSecret:    os.Getenv("GOOGLE_CLIENT_SECRET"),
+		AuthURL:         "https://accounts.google.com/o/oauth2/v2/auth",
+		TokenURL:        "https://oauth2.googleapis.com/token",
```
Comment (gurnoor6) [inline backend/login/login.go:39]: Let's add these also to constants

**Review by gurnoor6 — APPROVED** [review]
Looks good! Added a couple of minor comments. We can merge after fixing these

**Pull request merged**